### PR TITLE
features: commandline interface, custom video background, with some other features, and improvements, as well as bug fixes

### PR DESCRIPTION
**Pull request by iaacornus**
used `argparse` for cli implementation, and complied to pep8 coding style.

## summary

Other new features include
1. cli implementation, however can't build as package (`.whl`) yet, but can be resolved as `alias RedditVideoMakerBot="python3 cli.py"`.
2. custom background as parameter with `-b`, accepts link (only youtube), and an already downloaded mp4 file.
3. custom subreddit as parameter that can be specific with `-S`, if not specified, the program will default to AskReddit, or as specified in `.env` file, thus I suggest to remove the subreddit variable since it would be unnecessary.
4. #225 's "Name the final video file as the topic of the reddit thread", although not really that, but the final video can be renamed with `-f` parameter, e.g. "-f hello_world.mp4", this would also allow the user to set the destination of the final video without tampering with `.env` file, this parameter is not necessary tho since if it is not given the program would default to the name of the thread.
5. include `-t` (`--thread`) parameter for custom thread links.
6. include `--setup` option in cli for setup.
7. ability to avoid already finished threads (threads that already have video created for 
them).
9. allow users to specify how many comments should be included with `-n` parameter, defaults to none, and would include as many as the video could., should deal with #302 

Other changes:
1. complied to pep8
2. code refactoring, remove unnecessary codes
3. bug fixes via exception handling such as #340 
4. better `stderr` that can help user this includes the reported prawcore errors, as well as youtubedl errors.
5. overall code improvements.

## program example

```
python cli.py -h or if aliased to RedditVideoMakerBot or whatever, would work with -h
usage: RedditVideoMakerBot [OPTIONS]

Create Reddit Videos with one command.

options:
  -h, --help            show this help message and exit
  -c, --create          Create a video (uses the defaults).
  -s SUBREDDIT, --subreddit SUBREDDIT
                        Specify a subreddit.
  -b BACKGROUND, --background BACKGROUND
                        Specify a video background for video (accepts link and file).
  -f FILENAME, --filename FILENAME
                        Specify a filename for the video.
  -t THREAD, --thread THREAD
                        Use the given thread link instead of random.
  -n NUMBER, --number NUMBER
                        Specify number of comments to include in the video.
  --setup, --setup      (Re)setup the program.
```

You need to update the description tho, just add a commit to this pr.

I want to note that if there is no specified parameter, the program would default to the variables in `.env`, for subreddit, it defaults to askreddit, for threads, it is randomized, and for background it uses the minecraft parkour video, and for the filename, it uses the thread title.

### CODE REVIEW

**Comment by iaacornus** [comment]
I suggest to do further testing, as there are some glitches that happened in my test

**Comment by iaacornus** [comment]
Some of the input parameters is working by now, and others should work. Although currently `setup.py` is not producing the desired output.

Screenshot from 2022-06-05 11-37-49

**Comment by iaacornus** [comment]
The parameter `-b` is now working to download youtube videos:
Screenshot from 2022-06-05 12-02-46

**Comment by iaacornus** [comment]
some preview, custom background with the default askreddit

https://www.mediafire.com/file/33leorh1qfz2fxj/what's+the+scariest+thing+you've+heard+from+someone+who+talks+in+their+sleep.mp4/file

https://www.mediafire.com/file/wmnh9nkkb9tc6cl/Fewer+HP+Printer+Options+in+Linux+vs+Windows.mp4/file

**Comment by iaacornus** [comment]
I included the new updates in the master branch, so it would be compatible.

**Comment by iaacornus** [comment]
this commit should fix #300 #298 among others that suffers from OAuthExceptions

File: video_creation/final_video.py [hunk 1]
```diff
@@ -7,80 +11,86 @@
     CompositeAudioClip,
     CompositeVideoClip,
 )
-import reddit.subreddit 
-import re
-from utils.console import print_step
-from dotenv import load_dotenv
-import os
 
-W, H = 1080, 1920
```
Comment (CallumIO) [inline video_creation/final_video.py:16]: Thought process behind removing these variables? Surely it's more useful to keep them in.

**Review by CallumIO — COMMENTED** [review]
Looks mostly alright to me, waiting until next release and we move to a new branch

**Comment by gBlaku** [comment]
still not getting any comments on my vid. just the post

**Comment by iaacornus** [comment]
> still not getting any comments on my vid. just the post

I fixed it, you should be able to get comments now

**Review by Rohit2Lohia — APPROVED** [review]

**Comment by HallowedDust5** [comment]
Should another cli PR be made because this one has the stale label on it? Does the stable label indicate that the PR be closed? @CallumIO, I just want to know these details because I can probably rewrite the cli if need be.

**Comment by CallumIO** [comment]
> 

Sorry I missed this - if you'd like to, go for it. Alternatively, feel free to fix the merge conflicts on this

**Pull request closed without merging**